modules/shapefile/src/lib/parsers/parse-shapefile.js

```diff
@@ -5,26 +5,44 @@ import {zipBatchIterators} from '../streaming/zip-batch-iterators';
 import {SHPLoader} from '../../shp-loader';
 import {DBFLoader} from '../../dbf-loader';
 
+// eslint-disable-next-line max-statements
 export async function* parseShapefileInBatches(asyncIterator, options, context) {
+  const {metadata} = options;
```

Comment:
Can I assume that `options` is always an `object`, or is it ever `undefined`?

Reply:
Yes - If called from loaders.gl core it should always be an object. 

(If parser is called directly we cannot control it.)